server/app/views/applicant/AuthenticateUpsellViews.java

```diff
@@ -0,0 +1,73 @@
+package views.applicant;
+
+import static j2html.TagCreator.div;
+
+import controllers.routes;
+import j2html.tags.specialized.ButtonTag;
+import j2html.tags.specialized.DivTag;
+import java.util.Optional;
+import java.util.UUID;
+import play.i18n.Messages;
+import services.MessageKey;
+import views.BaseHtmlView;
+import views.components.ButtonStyles;
+import views.components.Modal;
+import views.components.Modal.Width;
+import views.style.StyleUtils;
+
+final class AuthenticateUpsellViews extends BaseHtmlView {
```

Comment:
**suggestion:** Should be AuthenticateUpsellView instead of AuthenticateUpsellViews, right?

Reply:
We have button tags to that are used by other classes, so there are kind of multiple views. And the naming leaves things open for more views to be added. I'm 50-50 on it, so you can decide.

Reply:
Hmm I'm kind of torn too. Another option is we could follow the precedent of https://github.com/civiform/civiform/blob/main/server/app/views/admin/programs/ProgramFormBuilder.java and call this something like UpsellViewBuilder or something.

Reply:
Went for `AuthenticateUpsellCreator` since the methods are `createX()`